### ISSUE TITLE
Roslyn unable to evaluate local variables on Windows Forms

### ISSUE THREAD

**Issue by isadorasophia** [dotnet/roslyn#66109]
**Version Used**: Visual Studio 17.5

**Steps to Reproduce**:

1. Clone the project at https://github.com/micah686/BrokenLocalsApp and add a breakpoint at BreakPrompt.cs, line 32.
2. Start debugging
3. Click the button and stop at breakpoint

**Expected**
User is able to evaluate the value of `textSize` and `topSize`.

**Actual**
Local variables are not listed. If attempted to evaluate:
![image](https://user-images.githubusercontent.com/7958854/209246306-823cc998-c0c8-4645-8fab-a4c84a4616ce.png)

**Remarks**
The IL lists these variables in the constructor body, so the underlying issue is not obvious:
```
.method public hidebysig specialname rtspecialname 
        instance void  .ctor(string text) cil managed
{
  // Code size       158 (0x9e)
  .maxstack  5
  .locals init ([0] valuetype [System.Drawing]System.Drawing.Size topSize,
           [1] valuetype [System.Drawing]System.Drawing.Size textSize)
```

**Further details**
Issue was created from https://developercommunity.visualstudio.com/t/Locals-not-showing-up-when-debugging/10221711. According to customer, this was a regression from VS 2019.

**Comment by isadorasophia** [comment]
I believe this ticket is also a duplicate on this: https://developercommunity.visualstudio.com/t/CS0103-in-the-watch-window-for-local-var/10222843. There seems to be a regression evaluating locals on Windows Forms.

**Comment by isadorasophia** [comment]
Issue was also reported here: https://github.com/dotnet/winforms/issues/8354

**Comment by kirsan31** [comment]
This is regression in 17.**4** from 17.3.
The fix is very needed 🙏

**Comment by bowpay** [comment]
This is a problem with any Winforms 6.0 or 4.7.2 project.  This is a MAJOR problem that needs to be fixed before Christmas!!!  No presents for you!!

**Comment by cston** [comment]
It looks like the `DocumentHandle` returned from `MetadataReader.GetMethodDebugInformation()` is `nil` (see [MethodDebugInfo.Portable.cs](https://github.com/dotnet/roslyn/blob/main/src/ExpressionEvaluator/Core/Source/ExpressionCompiler/PDB/MethodDebugInfo.Portable.cs#L66)) which results in an exception reading the debug information for the method.

**Comment by cston** [comment]
cc @RikkiGibson, @tmat

**Comment by cston** [comment]
The bug appears to affect methods with executable code from multiple documents, such as a constructor from a type with field initializers in multiple partial classes.

**Comment by kirsan31** [comment]
**17.5** - really? Is it possible to service this bug? 🙏
It's a really nightmare to work without locals debugging 😭

**Comment by jaredpar** [comment]
This is being considered for servicing in 17.4 but we need to track down some flakiness issues before we can push it through.

**Comment by kirsan31** [comment]
Fixed in 17.4.**5**.
